x/ecosystem-incentive/keeper/incentive_unit_service.go

```diff
@@ -37,6 +37,18 @@ func (k Keeper) Register(ctx sdk.Context, msg *types.MsgRegister) (*[]types.Subj
 		return nil, err
 	}
 
+	// operation related to IncentiveUnitIdsByAddr
+	// if exists already, add incentuve unit id in msg into data
+	// if not, newly create and set
+	for _, addr := range msg.SubjectAddrs {
+		incentiveUnitIdsByAddr := k.GetIncentiveUnitIdsByAddr(ctx, addr.AccAddress())
+		incentiveUnitIdsByAddr = incentiveUnitIdsByAddr.CreateOrUpdate(addr, msg.IncentiveUnitId)
+
+		if err := k.SetIncentiveUnitIdsByAddr(ctx, incentiveUnitIdsByAddr); err != nil {
```

Comment:
What is the maximum incentive units count per address?

Reply:
does that validation make sense practically?
Users can change the address to create a unit even if there's such limitation, can't they?
even so, you think it has to have such limitation?